### ISSUE TITLE
`object-*-newline`: missing TypeScript version

### ISSUE THREAD

**Issue by Ragnar-Oock** [eslint-stylistic/eslint-stylistic#228]
### Validations

- [X] Follow our [Code of Conduct](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://eslint.style/contribute/guide).
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.

### Describe the bug

When using `@stylistic/object-curly-newline` or `@stylistic/object-property-newline` such as :

```js
/*...*/
    rules: {
        '@stylistic/object-curly-newline': [ 'warn', {
            multiline: true,
            minProperties: 1,
            consistent: true
        } ],
        '@stylistic/object-property-newline': [ 'warn' ]
    }
/*...*/
```

The types declared either as `type TypeName = {...}` or directly as a nested property are not affected by the rules while object literals are.

### Reproduction

https://stackblitz.com/edit/eslint-stylistic-eslint-stylistic-u7qmmh?file=playground%2Findex.ts

### Contributes

- [ ] I am willing to submit a PR to fix this issue
- [ ] I am willing to submit a PR with failing tests

**Comment by Namchee** [comment]
Thses 2 rules are supports JS-syntax only since they aren't on [typescript-eslint ruleset](https://typescript-eslint.io/rules/). We need to create a new TypeScript-version of these rules.
